### PR TITLE
Update `CupertinoPicker` example

### LINKED ISSUE #110732: CupertinoPicker example does not select initial value when dismissed
Steps to reproduce:

1. go to https://api.flutter.dev/flutter/cupertino/CupertinoPicker-class.html
2. open the picker in the dartpad and select a value other than Apple, e.g. Orange
3. dismiss the picker and observe that the button has updated to the selected value
4. open the picker again and observe that the initially selected value is Apple
5. dismiss the picker and observe that the button does not update to Apple

Expected behavior:
the button should be updated to the selected value when the picker is dismissed

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/110732

Minor improvements to the example and update the example test.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: examples/api/lib/cupertino/picker/cupertino_picker.0.dart [hunk 1]
```diff
@@ -37,7 +38,7 @@ class CupertinoPickerExample extends StatefulWidget {
 }
 
 class _CupertinoPickerExampleState extends State<CupertinoPickerExample> {
-  int _selectedFruit = 0;
+  int selectedFruit = 0;
```
Comment (gspencergoog) [inline examples/api/lib/cupertino/picker/cupertino_picker.0.dart:41]: If you wouldn't make this a public accessor if the class were public, then don't make it a public accessor here. I know, this is an example, but it's what we'd do in production code.

File: examples/api/lib/cupertino/picker/cupertino_picker.0.dart [hunk 2]
```diff
@@ -6,8 +6,8 @@
 
 import 'package:flutter/cupertino.dart';
 
-const double _kItemExtent = 32.0;
-const List<String> _fruitNames = <String>[
+const double kItemExtent = 32.0;
```
Comment (gspencergoog) [inline examples/api/lib/cupertino/picker/cupertino_picker.0.dart:9]: These should probably stay private.

File: examples/api/lib/cupertino/picker/cupertino_picker.0.dart (hunk 1 above)
Comment (TahaTesser) [reply examples/api/lib/cupertino/picker/cupertino_picker.0.dart:41]: That makes sense. Thank you

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**